### PR TITLE
refactor: create a common resource to store debounce consts

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #5371

## Summary
Refactor: create a common resource to store debounce consts for `action-bar`, `combobox`, `filter`, `list`, `floating-ui`.

### CODE REVIEW

File: packages/calcite-components/src/utils/debounceValues.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// UI Reposition Debounce
+// - UseCase: In scenarios where multiple reposition requests might be made in quick succession (e.g., due to rapid UI changes), calling debounce helps in optimizing performance by reducing unnecessary computations and re-renderings.
+// - Use in the Component:
+
+// Floating-ui `positionFloatingUI` is called with debounce.
+export const repositionDebounceInMs = 100;
+
+// Drag Debounce
+// UseCase: Methods can fire frequently during drag. To perform expensive operations consider using a debounce to avoid locking up the main thread.
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/debounceValues.ts:31]: This one seems to be missing.

**Review by jcfranco — CHANGES_REQUESTED** [review]
This is headed in the right direction, but let's try to:

* move shared debounce delays to `calcite-components/src/utils/resources.ts`
* consolidate values by context (e.g., filtering, positioning)

File: packages/calcite-components/src/utils/debounceValues.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Resizing Debounce
+// - Usecase: The `debounce` function ensures that the `resize` method is not called more often than the specified milliseconds, regardless of how frequently the resize event is triggered.
+// - Use in the Component:
+
+// Action-bar resize method is called with debounce when it overflows actions that won't fit into menus.
+export const overflowActionsDebounceInMs = 150;
+export const overflowActionsDebounceInMsExtended = overflowActionsDebounceInMs + 10;
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/utils/debounceValues.ts:7]: Agreed. we probably don't need to mess with e2e ones.

**Comment by driskull** [comment]
Agreed.

Something like:

```ts
export const debounceValues = {
  reposition: 100,
  filter: 250,
  overflow: 150,
  nextTick: 0
};
```

File: packages/calcite-components/src/utils/resources.ts [hunk 3]
```diff
@@ -41,3 +41,10 @@ export const CSS_UTILITY = {
 export const TEXT = {
   loading: "Loading",
 };
+
+export const debounceValues = {
+  filter: 250,
+  nextTick: 0,
+  overflow: 150,
```
Comment (driskull) [inline packages/calcite-components/src/utils/resources.ts:48]: I wonder if we should rename `overflow` to `resize`?

Comment (Elijbet) [reply packages/calcite-components/src/utils/resources.ts:48]: Yeah, in this case it's wrapping the resize method.

Comment (driskull) [reply packages/calcite-components/src/utils/resources.ts:48]: if its wrapping the resize method, maybe it can just use the resize value?

Comment (driskull) [reply packages/calcite-components/src/utils/resources.ts:48]: Sounds good. Lets rename to `resize` 👍

File: packages/calcite-components/src/components/action-bar/utils.ts [hunk 4]
```diff
@@ -3,7 +3,6 @@ import { SLOTS as ACTION_GROUP_SLOTS } from "../action-group/resources";
 import { SLOTS as ACTION_MENU_SLOTS } from "../action-menu/resources";
 import { Layout } from "../interfaces";
 
-export const overflowActionsDebounceInMs = 150;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/action-bar/utils.ts:6]: ✨🧹✨

File: packages/calcite-components/src/utils/resources.ts [hunk 5]
```diff
@@ -41,3 +41,10 @@ export const CSS_UTILITY = {
 export const TEXT = {
   loading: "Loading",
 };
+
+export const debounceValues = {
```
Comment (jcfranco) [inline packages/calcite-components/src/utils/resources.ts:45]: For consistency, this should use `CONSTANT_CASE`. `Values` is implied, so WDYT about `DEBOUNCE`?

File: packages/calcite-components/src/utils/resources.ts (hunk 3 above)
Comment (jcfranco) [reply packages/calcite-components/src/utils/resources.ts:48]: I dig it!

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts [hunk 6]
```diff
@@ -1733,11 +1748,30 @@ describe("calcite-combobox", () => {
...
       await input.click();
       await input.press("o");
+
+      await page.waitForChanges();
+      await page.waitForTimeout(DEBOUNCE.filter);
+
       await input.press("Tab");
+
+      await page.waitForChanges();
+      await page.waitForTimeout(DEBOUNCE.filter);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox/combobox.e2e.ts:1758]: As a follow-up, let's see if this extra pair of `waitForX`s are needed. Also, some of the E2E/Puppeteer APIs call `waitForChanges` under the scenes, so some calls may not be necessary.

**Review by jcfranco — APPROVED** [review]
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉
🎉🥳🎉🎉🎉🥳🎉🎉🥳🥳🎉🎉🥳🥳🥳🎉🎉🥳🥳🥳🎉🥳🥳🥳🥳🎉🥳🎉
🎉🥳🥳🎉🎉🥳🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🎉🎉🥳🎉🎉🎉🎉🥳🎉
🎉🥳🎉🥳🎉🥳🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🎉🎉🥳🥳🥳🎉🎉🥳🎉
🎉🥳🎉🎉🥳🥳🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🥳🎉🎉🎉🎉🥳🎉🎉🎉🎉🎉🎉
🎉🥳🎉🎉🎉🥳🎉🎉🥳🥳🎉🎉🥳🥳🥳🎉🎉🥳🥳🥳🎉🥳🥳🥳🥳🎉🥳🎉
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

File: packages/calcite-components/src/components/combobox/combobox.e2e.ts (hunk 6 above)
Comment (Elijbet) [reply packages/calcite-components/src/components/combobox/combobox.e2e.ts:1758]: Sounds good, I'm removing that waitForX double, seem so to pass without it.

**Pull request merged**